src/core/task/tools/handlers/BrowserToolHandler.ts

```diff
@@ -112,15 +112,8 @@ export class BrowserToolHandler implements IFullyManagedTool {
 				await config.callbacks.say("browser_action_result", "")
 
 				// Re-make browserSession to make sure latest settings apply
-				const browserSession = config.services.browserSession
-				if (config.context) {
-					await browserSession.dispose()
-					const apiHandlerModel = config.api.getModel()
-					const useWebp = config.api ? !modelDoesntSupportWebp(apiHandlerModel) : true
-					config.services.browserSession = new BrowserSession(config.context, config.browserSettings, useWebp)
```

Comment:
This led to a bug where after Cline launched the browser at a URL, any subsequent calls to that browser session failed because the config in config.services.browserSession is a local object